common/lib/xmodule/xmodule/assets/vertical/public/js/vertical_student_view.js

```diff
@@ -1,16 +1,47 @@
 /* JavaScript for Vertical Student View. */
-window.VerticalStudentView = function(runtime, element) {
-    'use strict';
-    RequireJS.require(['course_bookmarks/js/views/bookmark_button'], function(BookmarkButton) {
-        var $element = $(element);
-        var $bookmarkButtonElement = $element.find('.bookmark-button');
 
-        return new BookmarkButton({
-            el: $bookmarkButtonElement,
-            bookmarkId: $bookmarkButtonElement.data('bookmarkId'),
-            usageId: $element.data('usageId'),
-            bookmarked: $element.parent('#seq_content').data('bookmarked'),
-            apiUrl: $bookmarkButtonElement.data('bookmarksApiUrl')
-        });
+import BookmarkButton from 'course_bookmarks/js/views/bookmark_button.js';
+
+class VerticalStudentView {
+  constructor(runtime, element) {
+    this.bookmarkButton = this.createBookmarkButton(element);
+    this.registerCompletions(runtime, element);
+  }
+
+  static createBookmarkButton(element) {
+    const $element = $(element);
+    const $bookmarkButtonElement = $element.find('.bookmark-button');
+    const bookmark = new BookmarkButton({
+      el: $bookmarkButtonElement,
+      bookmarkId: $bookmarkButtonElement.data('bookmarkId'),
+      usageId: $element.data('usageId'),
+      bookmarked: $element.parent('#seq_content').data('bookmarked'),
+      apiUrl: $bookmarkButtonElement.data('bookmarksApiUrl'),
     });
-};
+    return bookmark;
+  }
+
+  static registerCompletions(runtime, element) {
+    const $verts = $(element).find('.vert');
+    $verts.each(
+      (idx, completable) => {
+        if (completable.dataset.completable === undefined) {
```

Comment:
Two possible (untested) name changes captured below. 
`completable.dataset.completable` => `block.dataset.completableByViewing`

See [stackoverflow post](https://stackoverflow.com/questions/22753629/jquery-get-html-5-data-attributes-with-hyphens-and-case-sensitivity) for where I took naming conversion from `data-completable-by-viewing` => `dataset.completableByViewing`.